docs/DeveloperGuide.md

```diff
@@ -1,34 +1,245 @@
 # Developer Guide
 
-## Design & implementation
+## 1. Introduction
+Cent Wise Dollar Wise is a desktop application for money management, optimised for use via a Command Line Interface (CLI). This application aims to help NUS hostel students manage their financial situation and track their spending efficiently. 
+This developer guide provides information on the architecture and design of the application. It will not only help you get started as a Cent Wise Dollar Wise contributer, but that you will find useful to refer to even if you are already a contributer of this project. 
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## 2. Setting up
+### 2.1 Prerequisites
+- JDK 1.8.0_60 or later </br>
+🛈 Having any Java 8 version is not enough. </br>
+🛈 This app will not work with earlier versions of Java 8.
+- IntelliJ IDE </br>
+🛈 IntelliJ by default has Gradle and JavaFx plugins installed. </br>
+🛈 Do not disable them. If you have disabled them, go to File > Settings > Plugins to re-enable them.
+### 2.2 Setting up the project in your computer
+1. Fork this repo, and clone the fork to your computer.
+2. Open IntelliJ. If you are not in the welcome screen, click File > Close Project to close the existing project dialog first.
+3. Set up the correct JDK version for Gradle.
+4. Click Configure > Project Defaults > Project Structure.
+5. Click New…​ and find the directory of the JDK.
+6. Click Import Project.
+7. Locate the build.gradle file and select it. Click OK
+8. Click Open as Project
+9. Click OK to accept the default settings
+10. Open a console and run the command gradlew processResources (Mac/Linux: ./gradlew processResources). It should finish with the BUILD SUCCESSFUL message. This will generate all resources required by the application and tests.
+### 2.3 Verifying the setup
+1. Run the seedu.duke.Duke and try a few commands.
+2. Run the tests and ensure all the tests pass.
 
+## 3. Design & implementation
+### 3.1 Architecture
+![image](https://user-images.githubusercontent.com/45732128/97327381-e083a780-18af-11eb-8fed-eb6d7de73703.png) </br>
+Figure 1: Architecture diagram
```

Comment:
The architecture diagram seems a little messy as there is no clear structure to the diagram, might want to look into having slight changes to the diagram layout